home-accountant-web/src/main/java/com/godeltech/ha/service/MavenVersionService.java

```diff
@@ -0,0 +1,33 @@
+package com.godeltech.ha.service;
+
+import com.godeltech.ha.model.MavenVersion;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+
+
+import java.io.FileReader;
+import java.io.IOException;
+
+
+public class MavenVersionService {
+
+private MavenVersion mavenVersion = new MavenVersion();
```

Comment:
avoid using fields in Singleton - it shouldn't keep state (recommend to read about singleton pattern).
MavenVersion object can be created right in the method and then returned.

Reply:
why does Singleton shouldn't keep state?